### PR TITLE
superpmi.py script improvements

### PR DESCRIPTION
**Pull request by BruceForstall**
1. Make the script pylint and pycodestyle clean (with exclusions).
Add pylint/pycodestyle configuration files.

2. Simplify and remove duplication in asm diffs code for generating
asm and JIT dump files.

3. Add cases to automatically upload and download associated .MCT files
if one or more specific .MCH files are specified.

4. Fix a few bugs related to using VS Code to view diffs. (I'm inclined
to remove this complication entirely unless someone actually uses it.)

### CODE REVIEW

**Comment by BruceForstall** [comment]
@kunalspathak PTAL
cc @dotnet/jit-contrib

File: src/coreclr/scripts/.pylintrc [hunk 1]
```diff
@@ -0,0 +1,31 @@
+# Settings for pylint
+
+[MESSAGES CONTROL]
+
+# Disable the message, report, category or checker with the given id(s). You
```
Comment (kunalspathak) [inline src/coreclr/scripts/.pylintrc:5]: With this file, does this show errors/warnings in VS Code? I might have to run this on the new file `superpmi-setup.py` that I am adding.

File: src/coreclr/scripts/superpmi.py [hunk 2]
```diff
@@ -185,7 +176,7 @@
 
 merge_mch_pattern_help = """\
 A pattern to describing files to merge, passed through directly to `mcs -merge`.
-Acceptable patterns include `*.mch`, `file*.mch`, and `c:\my\directory\*.mch`.
+Acceptable patterns include `*.mch`, `file*.mch`, and `c:\\my\\directory\\*.mch`.
```
Comment (kunalspathak) [inline src/coreclr/scripts/superpmi.py:179]: Probably worth mentioning that it does recursive search inside `c:\\my\\directory` with the pattern as it was not obvious when I first saw the help message.

File: src/coreclr/scripts/superpmi.py [hunk 3]
```diff
@@ -468,52 +468,69 @@ def run_and_log(command, log_level=logging.DEBUG):
...
+    for line in stdout_output.decode('utf-8').splitlines():  # There won't be any stderr output since it was piped to stdout
         logging.log(log_level, line)
     return proc.returncode
 
 ################################################################################
 # Helper classes
 ################################################################################
 
+
 class TempDir:
```
Comment (kunalspathak) [inline src/coreclr/scripts/superpmi.py:483]: Some day, worth moving common parts into `utils.py` so we can reuse it.

File: src/coreclr/scripts/superpmi.py [hunk 4]
```diff
@@ -1285,168 +1343,53 @@ def replay_with_asm_diffs(self):
...
-                            assert diff_txt is not None
-                            assert diff_txt != ""
-
-                            if base_txt != diff_txt:
-                                text_differences.put_nowait(item)
-                    ################################################################################################ end of create_asm()
-
-                    async def create_jit_dump(print_prefix, item, self, mch_file, jit_dump_differences, base_dump_location, diff_dump_location):
-                        """ Run superpmi over an mc to create JIT dumps for the method.
+                    async def create_replay_artifacts(print_prefix, item, self, mch_file, env_vars, jit_differences_queue, base_location, diff_location, extension):
```
Comment (kunalspathak) [inline src/coreclr/scripts/superpmi.py:1355]: there is usually a rule for long lines.

**Review by kunalspathak — APPROVED** [review]
I should admit that I just skimmed through the pslint changes and they LGTM.

File: src/coreclr/scripts/superpmi.py (hunk 4 above)
Comment (BruceForstall) [reply src/coreclr/scripts/superpmi.py:1355]: Yes, I disabled the long lines rule... too many cases :-)

**Pull request merged**